diff --git a/.gitignore b/.gitignore
index df09924..04a6b37 100644
--- a/.gitignore
+++ b/.gitignore
@@ -170,4 +170,4 @@ cython_debug/
 # PyPI configuration file
 .pypirc
 /test-tmp
-/report_new_linter_errors
+/report-new-linter-errors
diff --git a/example/delete-previous.py b/example/delete-previous.py
index 75b4aef..a8ea2ad 100644
--- a/example/delete-previous.py
+++ b/example/delete-previous.py
@@ -1,4 +1,3 @@
 def foo(i: int) -> str:
     return f"foo{i}"
 
-
diff --git a/report_new_linter_errors.py b/report_new_linter_errors.py
index dd40ec0..34d616b 100644
--- a/report_new_linter_errors.py
+++ b/report_new_linter_errors.py
@@ -6,7 +6,8 @@ $ report-new-linter-errors <command> [args...]
 import os
 import shutil
 import subprocess
-from typing import Protocol, TextIO, Any, cast
+from dataclasses import dataclass
+from typing import Protocol, TextIO, Any, cast, Iterator, Literal, Optional
 
 # https://stackoverflow.com/questions/1101957/are-there-any-standard-exit-status-codes-in-linux
 EXIT_CODE_NO_SNAPSHOT_FILE = 66
@@ -35,20 +36,15 @@ class Sys(Protocol):
 
 
 def main(environ: Environ, sys: Sys) -> None:
-    snapshot_dir = environ.get(
-    )
-    if not os.path.exists(snapshot_dir):
-        os.makedirs(snapshot_dir)
+    snapshot_dir = SnapshotDirectory.from_environ(environ)
 
-    snapshot_path = os.path.join(snapshot_dir, 'snapshot')
+    snapshot_path = snapshot_dir.get_snapshot_path()
 
     with subprocess.Popen(linter_command, stdout=subprocess.PIPE,
                           stderr=subprocess.PIPE,
                           text=True, ) as linter_proc:
-        new_snapshot_path = os.path.join(snapshot_dir, 'new-snapshot')
+        new_snapshot_path = snapshot_dir.get_new_snapshot_path()
         with open(new_snapshot_path, 'w') as f:
             while True:
                 chunk = linter_proc.stdout.read(4096)
diff --git a/test.py b/test.py
index 0ae8627..c52c083 100644
--- a/test.py
+++ b/test.py
@@ -147,5 +148,53 @@ class ReportNewLinterErrorsTestCase(unittest.TestCase):
                 "snapshot file is NOT updated",
             )
 
+
+    class AdjustLineNumbersTestCase(unittest.TestCase):
+
+        def test_adjust_line_numbers_increments_line_number(self):
+            pass
+
+
